Adding APIs for the new resource - Network Security Perimeter (NSP)

harics24 opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [X] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [X] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

harics24 committed:
Merge pull request #3 from Azure/master

Taking latest from upstream

harics24 committed:
Merge branch 'Azure:master' into master

harics24 committed:
Merge branch 'Azure:master' into nsp-resource

Commit:
Defining the new resource NSP and the APIs

jorgecotillo commented on specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkSecurityPerimeterGet.json:
```diff
@@ -0,0 +1,23 @@
+  },
+  "responses": {
+    "200": {
+      "body": {
```
>"body" [](http://example.com/codeflow?start=6&length=6)

Your get call also returns an "etag" property right? might be a good idea to include it here as well to prevent confusions (mainly because your other examples do include an etag property)

harics24 replied:
Added the same.

Commit:
Adding the tag in get to be consistent with the list

Commit:
Removing etags everywhere as per internal discussion

lirenhe commented:
@harics24, could you take a look at the tool check error?  seems you missed  the following endpoint:

The private endpoint API: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/privateLinkServices/{serviceName}/privateLinkResources is missing.
Location: Microsoft.Network/stable/2020-11-01/webapplicationfirewall.json#L36

harics24 committed:
Merge pull request #5 from Azure/master

Taking latest from parent's master

Commit:
Creating a new folder for a new preview version

Adding NSP resource changes

Commit:
Changing version in all the files

Commit:
Update readme.md

Commit:
Removing the changes in the old version

Commit:
Removing the network manager files

Planning to use this version just for NSP and related resources. NM won't work with this version.

Commit:
Update readme.md

Commit:
Update readme.md

Commit:
Fixing cross version breaking change

Commit:
Removing the required fields in the correct version

Update network.json

Commit:
Attempting to fix the lint errors

harics24 commented:
@kairu-ms , @lirenhe , 

I have now added the changes in new version.
Addressed some of the lint exceptions.

I'm not sure how to fix the remaining. I did not make any changes for Go SDK, still it's showing validation exceptions. Please help.

harics24 commented:
The error is pointing to this file:

_The private endpoint API: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/privateLinkServices/{serviceName}/privateLinkResources is missing.
Location: Microsoft.Network/stable/2021-02-01/webapplicationfirewall.json#L36_

I did not make any changes there. We do not own this service/endpoint.

harics24 committed:
Merge branch 'Azure:master' into nsp-resource

Commit:
Adding the type as object to fix lint errors

lirenhe commented:
/azp run

lirenhe commented:
/azp run

lirenhe commented:
All lint errors are not related to this PR, approved for Lint.
